library/blob.c

```diff
@@ -1482,6 +1507,21 @@ static int eblob_fill_write_control_from_ram(struct eblob_backend *b, struct ebl
 		eblob_dump_wc(b, key, wc, "eblob_fill_write_control_from_ram: ERROR-pread-index", err);
 		goto err_out_cleanup_wc;
 	}
+
+	err = __eblob_read_ll(wc->data_fd, &data_dc, sizeof(data_dc), ctl.data_offset);
+	if (err) {
+		eblob_dump_wc(b, key, wc, "eblob_fill_write_control_from_ram: ERROR-pread-data", err);
+		goto err_out_cleanup_wc;
+	}
+
+	/* mark entry removed if its headers from index and data are different */
+	if (memcmp(&dc, &data_dc, sizeof(dc))) {
+		err = -EINVAL;
```

Comment:
What do you think about removing this object from index and data if headers mismatch?